src/jsonrpc-virtual-machine.cpp

```diff
@@ -223,11 +223,18 @@ void jsonrpc_mg_mgr::commit() {
         return;
     }
 
+    const std::string parent_address = m_address[0];
+
     // To commit, we kill the parent server and replace its address with the child's
     bool result = false;
     jsonrpc_request(get_mgr(), get_remote_parent_address(), "shutdown", std::tie(), result);
     std::swap(m_address[0], m_address[1]);
```

Comment:
Isn't this a pretty complicated logic for what should simply be a pop_back? Maybe I misread...

Reply:
This is exactly the same logic as before, I just added the few lines below. I did not simplify in case the next rebind request throws.

Reply:
I understand. But if the rebind throws, it's difficult to recover anyway, right? It may or may not have changed address before throwing. So we now need to figure out which one we should use to contact it. I think the only way is to try both?

Reply:
I simplified and just left a pop_back(). Continue to use the connection after a throw here is undefined behavior :) and the caller should give up on this connection, moreover the remote server could left zombie servers.